src/Stratis.Bitcoin.IntegrationTests/RPC/RPCTestsMutable.cs

```diff
@@ -12,6 +12,38 @@ namespace Stratis.Bitcoin.IntegrationTests.RPC
     /// </summary>
     public class RPCTestsMutable
     {
+        [Fact]
+        public void TestRpcGetBalanceIsSuccessful()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                CoreNode node = builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD().WithWallet();
+                builder.StartAll();
+                RPCClient rpcClient = node.CreateRPCClient();
+
+                TestHelper.MineBlocks(node, 2);
+                TestHelper.WaitLoop(() => node.FullNode.GetBlockStoreTip().Height == 2);
+
+                Money balance = rpcClient.GetBalance();
+                Assert.NotEqual(Money.Zero, balance);
```

Comment:
Funny you say that because I was surprised it passed as well. My expectations were the same as you. It actually returns a huge big balance 10000000 or something like that. Maybe it's a bug in the wallet or mining code? I'll step through the code and see where it's getting that from.

Reply:
this may be a bug in the wallet? or perhaps the wrong methods are being called to get balance.
 
@bokobza any thoughts?

Reply:
yeah that's fine. The GetBalance show the balance, which is correct in this case.
It's the GetSpendableTransactionsInAccount method that checks for maturity. This one is called when constructing a transaction. 
I agree that the names are close but the functionality is fine.

Reply:
Dug a bit deeper into this and it looks like bitcoin core returns the mature coins balance not confirmed balance. So I will be doing another PR to address that. I guess I need to change the RPC method to call GetSpendableTransactionsInAccount() instead. Good catch @zeptin !!